topics/add-dependencies.md

```diff
@@ -252,6 +429,55 @@ import MyFramework.*
 Learn more about [Objective-C and Swift interop](https://kotlinlang.org/docs/reference/native/objc_interop.html) and 
 [configuring cinterop from Gradle](https://kotlinlang.org/docs/reference/mpp-dsl-reference.html#cinterops).
 
+### Workaround to enable IDE support for the shared iOS source set {initial-collapse-state="collapsed"}
+
+Due to the [known issue](https://youtrack.jetbrains.com/issue/KT-40975), you won't be able to use IDE features, such as code completion and  highlighting, for the shared iOS source set 
+in a multiplatform project with [hierarchical structure support](https://kotlinlang.org/docs/reference/mpp-share-on-platforms.html#share-code-on-similar-platforms), if your project depends on:
+
+* Multiplatform libraries that don't support the hierarchical structure.
+* Third-party iOS libraries, except for [platform libraries](https://kotlinlang.org/docs/native-platform-libs.html) supported out of the box.
+
+This issue applies only to the shared iOS source set. The IDE will correctly support all the other code.
+
+> All KMM projects created with the KMM Project Wizard support the hierarchical structure and can also have this issue.
+>
+{type="note"}
+
+To enable IDE support in these cases, you can work around the issue and add the following code to `build.gradle.(kts)` in the `shared` directory of your project:
+
+<tabs>
+
+```Groovy
+def iosTarget
+if (System.getenv("SDK_NAME")?.startsWith("iphoneos")) {
+    iosTarget = kotlin.&iosArm64
+} else {
+    iosTarget = kotlin.&iosX64
+}
+```
+
+```Kotlin
+val iosTarget: (String, KotlinNativeTarget.() -> Unit) -> KotlinNativeTarget =
+    if (System.getenv("SDK_NAME")?.startsWith("iphoneos") == true)
+        ::iosArm64
+    else
+        ::iosX64
+
+iosTarget("ios")
+```
+
+</tabs>
+
+In this code sample, configuration of iOS targets depends on the environment variable `SDK_NAME`, which is managed by Xcode. 
```

Comment:
Not sure about the meaning of this sentence. What is it for? Do you expect any actions or conclusions?

Reply:
This is an explanation of how this workaround works, I add it in order to reduce the number of magic 🧙🏻‍♂️ users would copy to their build files

Reply:
Yes, it's a question from users. We've added this so that users could understand what the code does – not just blindly copy it.

Reply:
Maybe explain _what_ this variable reflects, not just _where it comes from_? As an ios newbie, I don't feel that this sentence makes the hack much clearer. But I agree if you think that it's not for newbies.

Reply:
Let's keep this way. If we get questions from newbies, we'll provide more details. Don't want to overload for now.

Reply:
You can google in any case :) and find an answer.